src/Chapter4.hs

```diff
@@ -673,6 +705,22 @@ Specifically,
  ❃ Implement the function to convert Tree to list
 -}
 
+data BTree e = 
+  Leaf 
+  | Node e (BTree e, BTree e)
+
+instance Functor BTree where
+  fmap :: (a -> b) -> BTree a -> BTree b
+  fmap _ Leaf = Leaf
+  fmap f (Node e (l,r)) = Node (f e) ((fmap f l), (fmap f r))
+
+reverseTree :: BTree e -> BTree e
+reverseTree Leaf = Leaf
+reverseTree (Node e (l, r)) = Node e ((reverseTree l), (reverseTree r))
```

Comment:
For reverse, you actually need to swap subtrees. This function is not changing the Tree, as you here would return the same that you get 🌴

Reply:
Oops. Thanks for pointing it out. Fixed it now 😊